xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -1163,6 +1163,7 @@ glusterd_brickinfo_new_from_brick(char *brick, glusterd_brickinfo_t **brickinfo,
     glusterd_brickinfo_t *new_brickinfo = NULL;
     xlator_t *this = NULL;
     char abspath[PATH_MAX] = "";
+    int len = 0;
```

Comment:
nit: can we use int8 here, I think new_brickinfo->hostname and new_brickinfo->path won't be too long to need a 16 bit dump for length.

Reply:
Should be size_t actually.

Reply:
@mykaul This is more of a general doubt, doesn't size_t occupy more space than int8_t. If the length is less, should we stick with size_t as a standard or should we go for something like int8_t. I know it is a standard but still..

Reply:
for example here we are dealing with lengths which will be in range of 128 so, shouldn't we sort of go with the implementation specific logic..

Reply:
It does. But it's not an array or anything, it's just few bytes more, and you save a conversion (which takes no time really for the CPU, but translates to few more bytes in the code). It's usually better to align to natural sizes and avoid conversions which may hide bugs, and is easier to read as well.

If that was an array, or within a struct (which may imply padding or alignment or both), then that would have been a different issue.

Reply:
Modern CPU's are optimized for 32 and 64 bits values. CPU's use a technique known as "register renaming" (they have more internal physical registers than those accessible from instructions, so that they are able to start executing unrelated instructions using the same register even before the previous instruction has terminated by used two different instances of the same register). This improves out-of-order execution and parallelism.

However, when small (partial) registers are used, the processor needs to create dependencies between instructions that could have been executed independently just because they may use the uninitialized part of the register (or something like that. I don't know the exact details). This could reduce the performance. Additionally, more instructions may be needed to convert between data types, as Yaniv said.

So, in general, small variables are better to be declared at least as 32 bit values even if they would fit in less bits. Only when the space is a critical factor (for example in big arrays) it makes sense to use smaller types. But even in this case, it's recommended to use a 32 bit value when reading data from the array. This way the CPU converts the small value into a regular 32-bit value and avoids this problem.

Sorry for the low-level explanation, but I thought it could be interesting as a background knowledge.

Reply:
@xhernandez , thank you for the detailed explanation :)